### PR TITLE
fix foreground censor script

### PR DESCRIPTION
**Pull request by ahnitz**
Requires #2773  + A future PR that adds back the combined foreground_veto that isn't keyed by ifo.

### CODE REVIEW

File: bin/hdfcoinc/pycbc_foreground_censor [hunk 1]
```diff
@@ -30,48 +28,22 @@ logging.info('start')
...
-        segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
-        fsegs += [(segs.coalesce() + mvsegs.coalesce()).coalesce()]
-        names += [args.output_segment_name]
-        ifos += [ifo]
+    ifos = f.attrs['ifos'].split(' ')
+
+ifos, fsegs, names = [], [], []
+for ifo in ifos:
+    segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
+    fsegs += [(segs.coalesce() + vsegs.coalesce()).coalesce()]
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_foreground_censor]: the `+` operation already yields a coalesced segment list, so no need for the outer brackets and `coalesce()`.

File: bin/hdfcoinc/pycbc_foreground_censor [hunk 2]
```diff
@@ -30,48 +28,22 @@ logging.info('start')
...
-# Check if using input files from old (2-ifo) coinc code
-if 'segments/foreground_veto/start' in f:
-    start = f['segments/foreground_veto/start'][:]
-    end = f['segments/foreground_veto/end'][:]
-    vsegs = pycbc.events.start_end_to_segments(start, end)
+start = f['segments/foreground_veto/start'][:]
+end = f['segments/foreground_veto/end'][:]
+vsegs = pycbc.events.start_end_to_segments(start, end)
 
+if 'detector_1' in f.attrs:
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_foreground_censor]: can you reinstate the comment about `# Check if using input files from old (2-ifo) coinc code`

File: bin/hdfcoinc/pycbc_foreground_censor [hunk 3]
```diff
@@ -30,48 +28,22 @@ logging.info('start')
...
-        names += [args.output_segment_name]
-        ifos += [ifo]
-
-    if args.strict_coinc_time:
-        fsegs[0] = (fsegs[0].coalesce() + fsegs[1].coalesce()).coalesce()
-        fsegs[1] = fsegs[0]
-
+    ifos = [ifo1, ifo2]
 else:
-    # Version used for multi-ifo coinc code
```
Comment (tdent) [inline bin/hdfcoinc/pycbc_foreground_censor]: ditto for this comment, keep it in

**Comment by GarethDaviesGW** [comment]
> A future PR that adds back the combined foreground_veto that isn't keyed by ifo.

see https://github.com/gwastro/pycbc/pull/2784

**Comment by tdent** [comment]
--- Did *anyone* test this?  @ahnitz ??

**Comment by tdent** [comment]
`ifos, fsegs, names = [], [], []
for ifo in ifos:`

**Pull request merged**